WordPress/Classes/ViewRelated/Reader/Filter/FilterProvider.swift

```diff
@@ -114,30 +115,69 @@ extension ReaderSiteTopic {
     }
 
     private static func tableProvider(completion: @escaping (Result<[TableDataItem], Error>) -> Void) {
-        fetchFollowedSites(completion: { result in
+        let completionBlock: (Result<[ReaderSiteTopic], Error>) -> Void = { result in
             let itemResult = result.map { sites in
-                sites.map { topic in
+                sites.filter {!$0.isExternal}.map { topic in
                     return TableDataItem(topic: topic, configure: { cell in
                         cell.textLabel?.text = topic.title
                         cell.detailTextLabel?.text = topic.siteURL
                     })
                 }
             }
             completion(itemResult)
-        })
+        }
+
+        fetchStoredFollowedSites(completion: completionBlock)
+        fetchFollowedSites(completion: completionBlock)
     }
 
+    /// Fetch sites from remote service
+    ///
     private static func fetchFollowedSites(completion: @escaping (Result<[ReaderSiteTopic], Error>) -> Void) {
         let siteService = ReaderTopicService(managedObjectContext: ContextManager.sharedInstance().mainContext)
 
         siteService.fetchFollowedSites(success: {
-            completion(.success(siteService.allSiteTopics().filter { !$0.isExternal }))
+            completion(.success(siteService.allSiteTopics()))
         }, failure: { error in
-            let unknownRestAPIError = NSError(domain: WordPressComRestApiErrorDomain, code: -1, userInfo: nil)
-            completion(.failure(error ?? unknownRestAPIError))
             DDLogError("Could not sync sites: \(String(describing: error))")
+            completion(.failure(error ?? FollowedSitesConstants.remoteServiceError))
         })
     }
+
+    /// Fetch sites from Core Data
+    ///
+    private static func fetchStoredFollowedSites(completion: @escaping (Result<[ReaderSiteTopic], Error>) -> Void) {
+        do {
```

Comment:
It looks like the following is doing pretty much the same work as `siteService.allSiteTopics()` which is called by above in `fetchFollowedSites`.  Am I missing something?  It might be nice if `fetchFollowedSites` and `fetchStoredFollowedSites` could get the list of sites the same way.

Reply:
🤦 yea.... I didn't notice what `allSiteTopics()` was doing. I'll update it. Thanks!